File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
    void dfs(vector<int>& rightSideView, int currLevel, TreeNode* root) {
        if (!root) return;
        if (static_cast<int>(rightSideView.size()) == currLevel) {
            rightSideView.push_back(root->val);
        } else {
            rightSideView[currLevel] = root->val;
        }
        dfs(rightSideView, currLevel + 1, root->left);
        dfs(rightSideView, currLevel + 1, root->right);
    }
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> rightSideView;
        dfs(rightSideView, 0, root);
        return rightSideView;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def visitNode(self, node: TreeNode, maxBefore: int):
        if not node:
            return 0
        increment = 0
        if maxBefore <= node.val:
            increment += 1
            maxBefore = node.val
        return increment + self.visitNode(node.left, maxBefore) + self.visitNode(node.right, maxBefore)
    def goodNodes(self, root: TreeNode) -> int:
        return self.visitNode(root, root.val)